components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1741,12 +1747,36 @@ private InboundAuthenticationConfig getInboundAuthenticationConfig(int applicati
             returnList.add(inboundAuthenticationRequestConfig);
         }
         InboundAuthenticationConfig inboundAuthenticationConfig = new InboundAuthenticationConfig();
-        inboundAuthenticationConfig.setInboundAuthenticationRequestConfigs(returnList
-                                                                                   .toArray(new InboundAuthenticationRequestConfig[returnList
-                                                                                           .size()]));
+        inboundAuthenticationConfig.setInboundAuthenticationRequestConfigs(returnList.toArray(new
+                InboundAuthenticationRequestConfig[returnList.size()]));
         return inboundAuthenticationConfig;
     }
 
+    private String getConfigTypeFromProperties(int applicationId, Connection connection, int tenantID) throws SQLException{
```

Comment:
Avoid additional DB call. Get the SP properties before getting authenticator configs pass them to getInboundAuthenticationConfig() method.